oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/InvalidIndexTest.java

```diff
@@ -0,0 +1,197 @@
+        LuceneIndexProvider provider = new LuceneIndexProvider();
+        luceneIndexEditorProvider.setBlobStore(blobStore);
+
+        asyncIndexUpdate = new AsyncIndexUpdate("async", nodeStore, compose(newArrayList(
+                luceneIndexEditorProvider,
+                new NodeCounterEditorProvider()
+        )));
+        TrackingCorruptIndexHandler trackingCorruptIndexHandler = new TrackingCorruptIndexHandler();
+        trackingCorruptIndexHandler.setCorruptInterval(INDEX_CORRUPT_INTERVAL_IN_MILLIS, TimeUnit.MILLISECONDS);
+        asyncIndexUpdate.setCorruptIndexHandler(trackingCorruptIndexHandler);
+        return new Oak(nodeStore)
+                .with(new InitialContent())
+                .with(new OpenSecurityProvider())
+                .with((QueryIndexProvider) provider)
+                .with((Observer) provider)
+                .with(luceneIndexEditorProvider)
+                .with(new PropertyIndexEditorProvider())
+                .with(new NodeTypeIndexProvider())
+                .with(new PropertyIndexProvider())
+                .with(new PropertyIndexEditorProvider())
+                .createContentRepository();
+    }
+
+    private void runEmptyAsyncIndexerCyclesWithoutNewContent(long count) {
+        for (int i = 0; i < count; i++) {
+            asyncIndexUpdate.run();
+        }
+    }
+
+    private void runIndexerCyclesAfterEachNodeCommit(long count, boolean async) throws CommitFailedException {
+        for (int i = 0; i < count; i++) {
+            root.getTree("/").addChild("testNode" + i);
+            root.commit();
+            if (async) {
+                asyncIndexUpdate.run();
+            }
+        }
+    }
+
+    private boolean assertionLogPresent(List<String> logs, String assertionLog) {
+        for (String log : logs) {
+            if (log.equals(assertionLog)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String invalidJcrPrimaryTypeLog() {
+        return "jcr:primaryType of index " + invalidJcrPrimaryTypeIndexName + " should be oak:QueryIndexDefinition instead of " + invalidJcrPrimaryTypeForIndex;
+    }
+
+    /**
+     * Logs warning as index is not a valid index definition
+     */
+    @Test
+    public void loggingInvalidJcrPrimaryType() throws Exception {
+        Tree test1 = root.getTree("/").addChild(INDEX_DEFINITIONS_NAME).addChild(invalidJcrPrimaryTypeIndexName);
+        test1.setProperty("jcr:primaryType", invalidJcrPrimaryTypeForIndex, Type.NAME);
+        test1.setProperty("type", "lucene");
+        test1.setProperty("reindex", true);
+        test1.setProperty("async", "async");
+        root.commit();
+        runIndexerCyclesAfterEachNodeCommit(contentCount, true);
+        runEmptyAsyncIndexerCyclesWithoutNewContent(contentCount);
+
+        List<String> logs = customLogger.getLogs();
+        assertEquals(assertionLogPresent(logs, invalidJcrPrimaryTypeLog()), true);
```

Comment:
You could use `assertTrue` here and in the tests below.

```suggestion
        assertTrue(assertionLogPresent(logs, invalidJcrPrimaryTypeLog()));
```